arraycontext/pytest.py

```diff
@@ -31,9 +31,44 @@
 
 # {{{ pytest integration
 
-def pytest_generate_tests_for_pyopencl_array_context(metafunc):
+from arraycontext.impl.pyopencl import PyOpenCLArrayContext
+from arraycontext.impl.pytato import PytatoArrayContext
+import pyopencl as cl
+from pyopencl.tools import _ContextFactory
+from typing import List
+
+
+class _PyOpenCLArrayContextFactory(_ContextFactory):
+    def __call__(self):
+        ctx = super().__call__()
+        from arraycontext.impl.pyopencl import PyOpenCLArrayContext
+        return PyOpenCLArrayContext(cl.CommandQueue(ctx))
+
+    def __str__(self):
+        return ("<PyOpenCL array context factory for <pyopencl.Device '%s' on '%s'>"
+                % (self.device.name.strip(),
+                 self.device.platform.name.strip()))
+
+
+class _PytatoArrayContextFactory(_ContextFactory):
+    def __call__(self):
+        ctx = super().__call__()
+        from arraycontext.impl.pytato import PytatoArrayContext
+        return PytatoArrayContext(cl.CommandQueue(ctx))
+
+    def __str__(self):
+        return ("<Pytato array context factory for <pyopencl.Device '%s' on '%s'>"
+                % (self.device.name.strip(),
+                 self.device.platform.name.strip()))
+
+
+types_to_factories = {PyOpenCLArrayContext: _PyOpenCLArrayContextFactory,
+    PytatoArrayContext: _PytatoArrayContextFactory}
+
+
+def pytest_generate_tests_for_array_contexts(metafunc, actx_list=None) -> None:
```

Comment:
Sorry, this still isn't clear to me. Could you please provide more details about what should be returned, and how this would be used? @kaushikcfd came up with the following code that works with the current code, but this isn't what you are looking for I guess?:

```python
pytest_generate_tests = 
  lambda metafunc: pytest_generate_tests_for_array_contexts(metafunc, actx_list=[PytatoPyOpenCLArrayContext])
```

Reply:
Is #44 a potential solution for this?

Reply:
Yep, I think so.